### DISCUSSION TITLE
Did TikTok-Api use Forging API/Scrapy/PySpider/Beautiful Soup/Selenium to scrape Tiktok's data?

### DISCUSSION THREAD

**Discussion by sunboyGithub** [davidteather/TikTok-Api#1087] [Q&A]
I see David has a course about scraping data with Foraging API and Beautiful soup. Besides, there are some popular scraping packages such as Scrapy/PySpider/Beautiful Soup/Selenium. However, I didn't see those packages are required in requirements.txt. Besides, they are not imported anywhere in this package, so I'm assuming TikTok-APIs doesn't use Forging API/Scrapy/PySpider/Beautiful Soup/Selenium. Am I right? If so, is TikTok-APIs purely developed from scratch? Thank you!

**Comment by eliich** [comment]
This is a wrapper api that sends requests directly to backend. In order to generate the signature for the requests, it first launches a headless playwright browser. Once signature is generated browser is closed, and requests are sent until either end of script reached or errors out
